File: src/main/java/github/macro/build_info/gems/acquisition/RewardType.java
package github.macro.build_info.gems.acquisition;

import org.jetbrains.annotations.Nullable;

import java.util.Arrays;

/**
 * Created by Macro303 on 2020-Jul-14.
 */
public enum RewardType {
	QUEST,
	VENDOR;

	@Nullable
	public static RewardType value(String name) {
		return Arrays.stream(values()).filter(tag -> tag.name().replaceAll("_", " ").equalsIgnoreCase(name)).findFirst().orElse(null);
	}
}

File: src/main/kotlin/github/macro/build_info/gems/reward/RewardType.kt
package github.macro.build_info.gems.reward

/**
 * Created by Macro303 on 2020-Feb-27.
 */
enum class RewardType {
	QUEST,
	VENDOR;

	companion object {
		fun value(name: String): RewardType? = values().firstOrNull {
			it.name.replace("_", " ").equals(name, ignoreCase = true)
		}
	}
}